### PR TITLE
[linux] Add support for WebKit2GTK 2.36+ features

### LINKED ISSUE #1498: [linux] Add support for WebKitGTK 2.36+
### Is your feature request related to a problem? Please describe.

WebKitGTK 2.36+ brings support for several new features when processing requests from the frontend, like response status codes, headers, http methods.

This would be especially interesting to bring the `AssetsHandler` at almost the same feature level like the other platforms and allows it to be used for a new plugin concept.

### Describe the solution you'd like

Add support for WebKitGTK 2.36+ and implement available features on the `AssetsHandler`.

### Describe alternatives you've considered

_No response_

### Additional context

It should be discussed in which form we want to add the new minimal dependency.
- Just require WebKITGTK 2.36+ for any build.
- Make it configurable to also allow building with an older version at the cost of loosing the new features.

2.36.0 has been released on March 21st, 2022 so it might be that not all distributions have already updated to it. On the other hand it seems like SecurityUpdates are only provided for the last major minor release. But still distributions might track their own branch with backports.

### PR DESCRIPTION
**Pull request by stffabi**
Fixes #1498

### CODE REVIEW

File: website/docs/reference/options.mdx [hunk 1]
```diff
@@ -249,24 +249,24 @@ dynamically with an `http.Handler` or hook into the request chain with an `asset
...
-| Response Body           | ✅  | ✅  | ✅  |
-| Response Body Streaming | ❌  | ❌  | ✅  |
-| WebSockets              | ❌  | ❌  | ❌  |
-
-NOTE: Linux is currently very limited due to targeting a WebKit2GTK Version < 2.36.0. In the future some features will be
-supported by the introduction of WebKit2GTK 2.36.0+ support.
+| Feature                 | Win | Mac | Lin    |
+| ----------------------- | --- | --- | ------ |
+| GET                     | ✅  | ✅  | ✅      |
+| POST                    | ✅  | ✅  | ✅ [^1] |
```
Comment (stffabi) [inline website/docs/reference/options.mdx:255]: @leaanthony , @misitebao does anybody of you know if that footnotes do work with our docs?

File: v2/internal/frontend/desktop/linux/webkit2_36.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
+//go:build linux && webkit2_36
+
+package linux
+
+/*
+#cgo linux pkg-config: gtk+-3.0 webkit2gtk-4.0 libsoup-2.4
```
Comment (leaanthony) [inline v2/internal/frontend/desktop/linux/webkit2_36.go:6]: Is `libsoup` a new dependency that we need to add to our website and package manager code?

**Comment by leaanthony** [comment]
Confirmed working on `Ubuntu 22.04`. Nice job!

File: v2/internal/frontend/desktop/linux/webkit2_36.go (hunk 2 above)
Comment (stffabi) [reply v2/internal/frontend/desktop/linux/webkit2_36.go:6]: WebKit2GTK uses `libsoup` internally, so I think it should not be possible to have WebKit2GTK installed without this dependency. But maybe it's better to be on the safe side and add it.

I'll add that as soon as we have merged the CLI refactoring so it doesn't interfere with that. Are you fine with that?

File: website/docs/reference/options.mdx (hunk 1 above)
Comment (stffabi) [reply website/docs/reference/options.mdx:255]: Ah yeah, always forget about that 😄

File: v2/internal/frontend/desktop/linux/webkit2_36.go (hunk 2 above)
Comment (leaanthony) [reply v2/internal/frontend/desktop/linux/webkit2_36.go:6]: Yeah sure, though I suspect you are right: it's probably installed by webkit2gtk anyway so no need to specify it 👍

**Comment by leaanthony** [comment]
#2152 merged, you should be good to go with this.

**Comment by stffabi** [comment]
Awesome, thanks for approving 🙏

File: website/docs/reference/options.mdx (hunk 1 above)
Comment (leaanthony) [reply website/docs/reference/options.mdx:255]: Oh my bad! I wondered why it looked like nothing had happened. The superscript 1 is super hard to click and it jumps to the bottom of the page. How about a `*` next to is with some text right under it or maybe another column with `Linux (webkit 2.36+)` in it?

Comment (stffabi) [reply website/docs/reference/options.mdx:255]: Sorry already merged 😔, but will open another PR to improve that.

**Pull request merged**